pkg/sql/catalog/descs/collection.go

```diff
@@ -185,6 +185,8 @@ type Collection struct {
 	// numbers of descriptors.
 	allDescriptors []catalog.Descriptor
 
+	allDescriptorsByID map[descpb.ID]catalog.Descriptor
```

Comment:
Gratuitous bikeshed: I believe @rafiss 's comments could be addressed by getting rid of `allDescriptors` altogether and using `allDescriptorsByID` instead. This means rewriting `GetAllDescriptors` so that it returns an ordered slice, which should be fine. If it isn't, replace `allDescriptorsByID` with a `catalog.DescriptorIDSet` perhaps? Duplicate sources-of-truth make me uncomfortable, I guess.